ape_hardhat/provider.py

```diff
@@ -198,6 +210,33 @@ def is_connected(self) -> bool:
     def _test_config(self) -> TestConfig:
         return cast(TestConfig, self.config_manager.get_config("test"))
 
+    @cached_property
+    def _package_json(self) -> PackageJson:
+        json_path = self.project_folder / "package.json"
+
+        if not json_path.is_file():
+            return PackageJson()
+
+        return PackageJson(**json.loads(json_path.read_text()))
+
+    @cached_property
+    def _hardhat_plugins(self) -> List[str]:
+        plugins: List[str] = []
+
+        def package_is_plugin(package: str) -> bool:
+            return re.search(r"hardhat-[A-Za-z0-9-]+$", package) is not None
```

Comment:
Have any more info on that?

Reply:
Was curious so went looking.  [According to the Python docs](https://docs.python.org/3/howto/regex.html#module-level-functions):

> Under the hood, these functions simply create a pattern object for you and call the appropriate method on it. They also store the compiled object in a cache, so future calls using the same RE won’t need to parse the pattern again and again.

So performance wise it sounds the same to me.  That said, maybe making the compiled pattern a file-level const might be a little cleaner?

Reply:
But look at the last paragraph in the link you sent me:

```
"""
Should you use these module-level functions, or should you get the pattern and call its methods yourself? If you’re accessing a regex within a loop, pre-compiling it will save a few function calls. Outside of loops, there’s not much difference thanks to the internal cache.
"""
```

So I am saying do exactly that since this is getting called in a loop, let's pre-compile it.
But you are right, it is not a optimization because of the caching.

Reply:
> @unparalleled-js so that type check failure doesn't seem related to my PR (I didn't touch pyyaml at all). Is this something you'd want me to address in this PR?

Yeah latest mypy has been causing issues, you can pin in setup.py to fix this issue. That is what we did in Ape core.

Reply:
Hm, thanks.  Guess that part didn't register when I read it.  Now I'm wondering if a file-level constant or defining it within the function like your suggestion makes any difference when it comes to the compiler.  I have zero idea, though I'd probably favor the constant even though it's only used in one place.  Let me know if you have a preference.